service/worker/indexer/indexer_test.go

```diff
@@ -82,3 +85,130 @@ func TestNewIndexer(t *testing.T) {
 	indexer := NewIndexer(config, mockMessagingClient, mockESClient, processorName, testlogger.New(t), metrics.NewNoopMetricsClient())
 	assert.NotNil(t, indexer)
 }
+
+// TestIndexerStart tests the Start method of Indexer
+func TestIndexerStart(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	config := &Config{
+		ESProcessorNumOfWorkers:  dynamicconfig.GetIntPropertyFn(1),
+		ESProcessorBulkActions:   dynamicconfig.GetIntPropertyFn(10),
+		ESProcessorBulkSize:      dynamicconfig.GetIntPropertyFn(2 << 20),
+		ESProcessorFlushInterval: dynamicconfig.GetDurationPropertyFn(1 * time.Minute),
+		IndexerConcurrency:       dynamicconfig.GetIntPropertyFn(1),
+	}
+	mockConsumer := messaging.NewMockConsumer(ctrl)
+	mockConsumer.EXPECT().Start().Return(nil).Times(1)
+	messageChan := make(chan messaging.Message)
+	defer close(messageChan)
+	mockConsumer.EXPECT().Messages().Return((<-chan messaging.Message)(messageChan)).AnyTimes()
+	mockvisibiltyProcessor := NewMockESProcessor(ctrl)
+	mockvisibiltyProcessor.EXPECT().Start().Return().Times(1)
+
+	indexer := &Indexer{
+		config:              config,
+		esIndexName:         "test-index",
+		consumer:            mockConsumer,
+		logger:              log.NewNoop(),
+		scope:               metrics.NoopScope(metrics.IndexProcessorScope),
+		shutdownCh:          make(chan struct{}),
+		visibilityProcessor: mockvisibiltyProcessor,
+		msgEncoder:          defaultEncoder,
+	}
+	err := indexer.Start()
```

Comment:
let's stop the indexer in this function to avoid leaking goroutines in tests. also you can validate nothing is leaking via `defer goleak.VerifyNone(t)`

Reply:
Thanks, I have added the stop and goleak check.